sdk/eventhub/azure-eventhubs/azure/eventhub/aio/producer_async.py

```diff
@@ -168,19 +160,22 @@ def _on_outcome(self, outcome, condition):
         :return: an EventDataBatch instance
         :rtype: ~azure.eventhub.EventDataBatch
         """
-        max_size = kwargs.get("max_size", None)
-        partition_key = kwargs.get("partition_key", None)
+
+        @_retry_decorator
+        async def _wrapped_open(*args, **kwargs):
+            await self._open(**kwargs)
+
         if not self._max_message_size_on_link:
-            await self._open()
+            await _wrapped_open(self, timeout=self.client.config.send_timeout)
 
         if max_size and max_size > self._max_message_size_on_link:
             raise ValueError('Max message size: {} is too large, acceptable max batch size is: {} bytes.'
                              .format(max_size, self._max_message_size_on_link))
 
-        return EventDataBatch(max_size or self._max_message_size_on_link, partition_key)
+        return EventDataBatch(max_size=(max_size or self._max_message_size_on_link), partition_key=partition_key)
 
-    async def send(self, event_data, **kwargs):
-        # type:(Union[EventData, EventDataBatch, Iterable[EventData]], Union[str, bytes]) -> None
+    async def send(self, event_data, partition_key=None, timeout=None):
```

Comment:
`async def send(self, event_data, *, partition_key=None, timeout=None)`

Reply:
it is fixed.